invoices/clock_test.go

```diff
@@ -0,0 +1,75 @@
+package invoices
+
+import (
+	"sync"
+	"time"
+)
+
+// testClock can be used in tests to mock time.
```

Comment:
Really nice! May I suggest you to extend `clock.Clock` interface and use that as a common ground instead? Maybe also move this test clock in that package such that others can use it?

Reply:
I don't want to delay this PR with a discussion on where to put that package/file. Given the time those took in past prs I'd rather avoid it (reference class forecasting).